src/main/java/net/earthcomputer/clientcommands/command/AreaStatsCommand.java

```diff
@@ -0,0 +1,205 @@
+                                .then(argument("predicates", list(blockPredicate(), 1))
+                                        .executes(ctx -> areaStats(ctx.getSource(), getBlockPos(ctx, "pos1"), getBlockPos(ctx, "pos2"), getList(ctx, "predicates"))))
+                                .executes(ctx -> areaStats(ctx.getSource(), getBlockPos(ctx, "pos1"), getBlockPos(ctx, "pos2"), block -> !block.getDefaultState().isAir())))));
+    }
+
+    private static int areaStats(ServerCommandSource source, BlockPos pos1, BlockPos pos2, Predicate<Block> blockPredicate) throws CommandSyntaxException {
+        return areaStats(source, pos1, pos2, Collections.singletonList(blockPredicate));
+    }
+
+    private static int areaStats(ServerCommandSource source, BlockPos pos1, BlockPos pos2, List<Predicate<Block>> blockPredicates) throws CommandSyntaxException {
+        final ClientWorld world = MinecraftClient.getInstance().world;
+        final ChunkManager chunkManager = world.getChunkManager();
+        if (!chunkManager.isChunkLoaded(pos1.getX() >> 4, pos1.getZ() >> 4) || !chunkManager.isChunkLoaded(pos2.getX() >> 4, pos2.getZ() >> 4)) {
+            throw NOT_LOADED_EXCEPTION.create();
+        }
+
+        final Predicate<Block> blockPredicate = or(blockPredicates);
+
+        final long startTime = System.nanoTime();
+
+        final WorldChunk chunk1 = world.getWorldChunk(pos1);
+        final WorldChunk chunk2 = world.getWorldChunk(pos2);
+
+        final int minX, maxX, minZ, maxZ, minY, maxY;
+        if (pos1.getX() <= pos2.getX()) {
+            minX = pos1.getX();
+            maxX = pos2.getX();
+        } else {
+            minX = pos2.getX();
+            maxX = pos1.getX();
+        }
+        if (pos1.getZ() <= pos2.getZ()) {
+            minZ = pos1.getZ();
+            maxZ = pos2.getZ();
+        } else {
+            minZ = pos2.getZ();
+            maxZ = pos1.getZ();
+        }
+        if (pos1.getY() <= pos2.getY()) {
+            minY = pos1.getY();
+            maxY = pos2.getY();
+        } else {
+            minY = pos2.getY();
+            maxY = pos1.getY();
+        }
+
+        final BlockPos.Mutable mutablePos = new BlockPos.Mutable();
+
+        int blocks = 0;
+        int chunks = 0;
+
+        if (chunk1.getPos().equals(chunk2.getPos())) {
+            chunks++;
+            blocks = loop(minX, maxX, minZ, maxZ, minY, maxY, blocks, blockPredicate, chunk1, mutablePos);
+        } else if (chunk1.getPos().x == chunk2.getPos().x) {
+            chunks++;
+            final WorldChunk minMinChunk = world.getChunk(minX >> 4, minZ >> 4);
+            blocks = loop(minX, maxX, minZ, (minZ >> 4) * 16 + 15, minY, maxY, blocks, blockPredicate, minMinChunk, mutablePos);
+
+            chunks++;
+            final WorldChunk minMaxChunk = world.getChunk(minX >> 4, maxZ >> 4);
+            blocks = loop(minX, maxX, (maxZ >> 4) * 16, maxZ, minY, maxY, blocks, blockPredicate, minMaxChunk, mutablePos);
```

Comment:
`loop` could be changed to return the number of blocks in that chunk, then you can use `blocks +=` here.